locale/Makefile

```diff
@@ -1,41 +1,76 @@
 #
-# Makefile for PO merging and MO generation
+# Makefile for PO merging and MO generation. More info in the README.
 #
-
-POTFILE = katello.pot
-POFILES = $(shell find . -name '*.po')
+# make all-mo (default) - generate MO files
+# make check - check translations using translate-tool
+# make tx-update - download and merge translations from Transifex
+# make clean - clean everything
+#
+DOMAIN = katello
+VERSION = $(shell git describe --abbrev=0 --tags)
+POTFILE = $(DOMAIN).pot
+MOFILE = $(DOMAIN).mo
+POFILES = $(shell find . -name '$(DOMAIN).po')
 MOFILES = $(patsubst %.po,%.mo,$(POFILES))
 POXFILES = $(patsubst %.po,%.pox,$(POFILES))
+EDITFILES = $(patsubst %.po,%.edit.po,$(POFILES))
 
 %.mo: %.po
-	msgfmt -o $@ $<
+	mkdir -p $(shell dirname $@)/LC_MESSAGES
+	msgfmt -o $(shell dirname $@)/LC_MESSAGES/$(MOFILE) $<
+	touch $(shell dirname $@)/LC_MESSAGES/$(MOFILE)
 
 # Generate MO files from PO files
 all-mo: $(MOFILES)
 
 # Check for malformed strings
-# TODO - enable endwhitespace, endpunc, puncspacing, options filters
 %.pox: %.po
 	msgfmt -c $<
 	pofilter --nofuzzy -t variables -t blank -t urls -t emails -t long -t newlines \
-		-t options -t printf -t validchars --gnome $< > $@
+		-t endwhitespace -t endpunc -t puncspacing -t options -t printf -t validchars --gnome $< > $@
+	cat $@
 	! grep -q msgid $@
 
-check: $(POXFILES)
+%.edit.po:
+	touch $@
 
-# Merge PO files
-update-po:
-	for f in $(shell find ./ -name "*.po") ; do \
-		msgmerge -N --backup=none -U $$f ${POTFILE} ; \
-	done
+check: $(POXFILES)
 
 # Unify duplicate translations
 uniq-po:
 	for f in $(shell find ./ -name "*.po") ; do \
 		msguniq $$f -o $$f ; \
 	done
 
-# Remove all MO files
-clean:
-	-rm -f messages.mo
-	find . \( -name "*.mo" -o -name "*.pox" \) -exec rm -f '{}' ';'
+tx-pull: $(EDITFILES)
+	tx pull -f
+	for f in $(EDITFILES) ; do \
+		sed -i 's/^\("Project-Id-Version: \).*$$/\1$(DOMAIN) $(VERSION)\\n"/' $$f; \
+	done
+
+tx-update: tx-pull
+	@echo
+	@echo Run rake plugin:gettext[$(DOMAIN)] from the Foreman installation, then make -C locale mo-files to finish
+	@echo
+
+mo-files: $(MOFILES)
+	git add $(POFILES) $(POTFILE) ../locale/*/LC_MESSAGES
+	git commit -m "i18n - pulling from tx"
```

Comment:
I think it'd be better to make this optional or interactive, at least (y/n). Here's why: most of us are running this from a vagrant box; rake must be run on the vagrant box with the environment (gems, etc.) set up, but git commits must be run from the hypervisor in order to use the proper git user and config (gpg signing, etc.).

Reply:
This is  a copy/paste for the other makefiles. I would prefer to keep them consistent, but I dont do builds. lets ping @ares @lzap @ehelms @dLobatog for their takes on consistency vs the pain of the commit (which I hate btw)

Reply:
I think this is fine given when this is typically run. We can't be sure where a user is running their git code, whether they have it synced or are using it directly from Vagrant (which I do). So let's keep it as is and let releasers figure it out over time.